image_get/worker/test/test_worker.py

```diff
@@ -0,0 +1,219 @@
+        )
+
+    def consume(self, block=True):
+        if self.messages:
+            return self.messages.pop()
+        else:
+            return None
+
+    def commit_offsets(self):
+        pass
+
+
+class FakeImageResponse:
+    def __init__(self, status=200, corrupt=False):
+        self.status = status
+        self.corrupt = False
+
+    async def read(self):
+        # 1024 x 768 sample image
+        if self.corrupt:
+            location = 'test/corrupt.jpg'
+        else:
+            location = 'test/test_image.jpg'
+        with open(location, 'rb') as f:
+            return f.read()
+
+
+class FakeAioSession:
+    def __init__(self, corrupt=False):
+        self.corrupt = corrupt
+
+    async def get(self, url):
+        return FakeImageResponse(self.corrupt)
+
+
+class AioNetworkSimulatingSession:
+    """
+    It's a FakeAIOSession, but it can simulate network latency, errors,
+    and congestion. At 80% of its max load, it will start to slow down and occasionally
+    throw an error. At 100%, error rates become very high and response times slow.
+    """
+
+    class Load(Enum):
+        LOW = auto()
+        HIGH = auto()
+        OVERLOADED = auto()
+
+    # Under high load, there is a 1/5 chance of an error being returned.
+    high_load_status_choices = [403, 200, 200, 200, 200]
+    # When overloaded, there's a 4/5 chance of an error being returned.
+    overloaded_status_choices = [500, 403, 501, 400, 200]
+
+    def __init__(self, max_requests_per_second=10, fail_if_overloaded=False):
+        self.max_requests_per_second = max_requests_per_second
+        self.requests_last_second = deque()
+        self.load = self.Load.LOW
+        self.fail_if_overloaded = fail_if_overloaded
+
+    def record_request(self):
+        """ Record a request and flush out expired records. """
+        if self.requests_last_second:
+            while (self.requests_last_second
+                   and time.time() - self.requests_last_second[0] > 1):
+                self.requests_last_second.popleft()
+        self.requests_last_second.append(time.time())
+
+    def update_load(self):
+        original_load = self.load
+        utilization =\
+            len(self.requests_last_second) / self.max_requests_per_second
+        if utilization <= 0.8:
+            self.load = self.Load.LOW
+        elif 0.8 < utilization < 1:
+            self.load = self.Load.HIGH
+        else:
+            self.load = self.Load.OVERLOADED
+            if self.fail_if_overloaded:
+                assert False, "You DDoS'd the server!"
+        if self.load != original_load:
+            log.debug(f'Changed simulator load status to {self.load}')
+
+    def lag(self):
+        """ Determine how long a request should lag based on load. """
+        if self.load == self.Load.LOW:
+            wait = random.uniform(0.05, 0.15)
+        elif self.load == self.Load.HIGH:
+            wait = random.uniform(0.15, 0.6)
+        # Overloaded
```

Comment:
worker_1     | Traceback (most recent call last):
worker_1     |   File "worker/consumer.py", line 1, in <module>
worker_1     |     import worker.settings as settings
worker_1     | ModuleNotFoundError: No module named 'worker'
image_get_worker_1 exited with code 1